src/lib.rs

```diff
@@ -339,6 +339,10 @@ fn format_ast<F>(krate: &ast::Crate,
                  -> Result<(FileMap, bool), io::Error>
     where F: FnMut(&str, &mut StringBuffer) -> Result<bool, io::Error>
 {
+    if config.disable_all_formatting {
```

Comment:
You probably want this check in the caller, rather than here, since there might be line-based reformating as well as AST-based reformatting